### PR TITLE
FEATURE: Asynchronously load policy information for nodes

### PR DESCRIPTION
**Pull request by kitsunet**
Avoid evaluating policies when fetching node information from
server in order to improve performance of loading/display of nodes.

Policies for Documents are only fetched when the respective node is
selected in the tree. Policies for content nodes are fetched after
the page is loaded in the iframe.

Related: #1579
Fixes: #1646

### CODE REVIEW

**Comment by kitsunet** [comment]
This also changes usage of the Nodes.ADD action to Nodes.MERGE to retain loaded policy information that would be erased by ADD.

File: Classes/Service/NodePolicyService.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @return array the key is a Privilege class name; the value is "true" if privileges are configured for this class name.
+     * @Flow\CompileStatic
+     */
+    public static function getUsedPrivilegeClassNames($objectManager)
```
Comment (mstruebing) [inline Classes/Service/NodePolicyService.php]: typehint's would be cool I think

File: packages/neos-ui/src/Sagas/CR/Policies/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+test(`write tests`, () => {
+    expect(true).toBe(true);
```
Comment (mstruebing) [inline packages/neos-ui/src/Sagas/CR/Policies/index.spec.js]: :1st_place_medal:

Comment (dimaip) [reply packages/neos-ui/src/Sagas/CR/Policies/index.spec.js]: What's the point of these empty test files? We never write any tests anyways...

Comment (kitsunet) [reply packages/neos-ui/src/Sagas/CR/Policies/index.spec.js]: :D I just followed best practice 😆

File: Classes/Service/NodePolicyService.php (hunk 1 above)
Comment (kitsunet) [reply Classes/Service/NodePolicyService.php]: right, will do!

File: packages/neos-ui-redux-store/src/CR/Nodes/index.js [hunk 3]
```diff
@@ -67,6 +69,12 @@ const add = createAction(ADD, nodeMap => ({nodeMap}));
  */
 const merge = createAction(MERGE, nodeMap => ({nodeMap}));
 
+/**
+ * Specific action for node information coming from the guest frame,
+ * to be able to trigger more specific sagas based on that information.
```
Comment (dimaip) [inline packages/neos-ui-redux-store/src/CR/Nodes/index.js]: > "to be able to trigger more specific sagas"
These ain't cool. I'd rather fire an additional action to inform about that, or listen to one of the existing actions, if there are any

Comment (kitsunet) [reply packages/neos-ui-redux-store/src/CR/Nodes/index.js]: Mmm, maybe my explanation is bad? So far there is only merge and add which is very generic. We don't know where it comes from. In theory we could also use those BUT as we trigger a merge after we fetched policies that might result in a possible recursion? That's why I added this more specific action.

File: packages/neos-ui/src/Sagas/CR/Policies/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const {endpoints} = backend.get();
+    while (true) {
+        const action = yield take(actionTypes.CR.Nodes.MERGE_FROM_GUEST_FRAME);
+        const {nodeMap} = action.payload;
+
+        const nodesWithoutPolicies = Object.keys(nodeMap).filter(contextPath => {
+            const getNodeByContextPathSelector = selectors.CR.Nodes.makeGetNodeByContextPathSelector(contextPath);
+            const node = select(getNodeByContextPathSelector);
+            const policyInfo = $get('policy', node);
+            return (!policyInfo || policyInfo === false);
```
Comment (dimaip) [inline packages/neos-ui/src/Sagas/CR/Policies/index.js]: I don't understand this condition. Isn't it equal to `!policyInfo`?

**Pull request merged**